core/dbt/contracts/graph/unparsed.py

```diff
@@ -142,6 +138,62 @@ class HasConfig:
     config: Dict[str, Any] = field(default_factory=dict)
 
 
+NodeVersion = Union[str, float]
+
+
+@dataclass
+class UnparsedVersion(dbtClassMixin):
+    v: NodeVersion
+    defined_in: Optional[str] = None
+    description: str = ""
+    access: Optional[str] = None
+    config: Dict[str, Any] = field(default_factory=dict)
+    meta: Dict[str, Any] = field(default_factory=dict)
+    constraints: List[Dict[str, Any]] = field(default_factory=list)
+    docs: Docs = field(default_factory=Docs)
+    tests: Optional[List[TestDef]] = None
+    columns: Sequence[Union[dbt.helper_types.IncludeExclude, UnparsedColumn]] = field(
+        default_factory=list
+    )
+
+    def __lt__(self, other):
+        try:
+            v = type(other.v)(self.v)
+            return v < other.v
+        except ValueError:
+            try:
+                other_v = type(self.v)(other.v)
+                return self.v < other_v
+            except ValueError:
+                return str(self.v) < str(other.v)
```

Comment:
Cool! This behaves how I would expect if `latest_version` isn't set. Defaults to greatest (numeric) or alphabetically last (string).